src/App.jsx

```diff
@@ -14,33 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_GOODS_ALPHABET = 'Alphabet';
+const SORT_GOODS_LENGTH = 'Length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortFild, reversed) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  preparedGoods.sort((good1, good2) => {
+    switch (sortFild) {
+      case SORT_GOODS_ALPHABET:
+        return good1.localeCompare(good2);
+      case SORT_GOODS_LENGTH:
+        return good1.length - good2.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortFild, setSortFild] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortFild, reversed);
+  const isChoice = sortFild || reversed;
+
+  const resetFild = () => {
+    setSortFild('');
+    setReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          // className="button is-info is-light"
+          className={cn(
+            `button is-info
+            ${sortFild === SORT_GOODS_ALPHABET || 'is-light'} `,
+          )}
+          onClick={() => setSortFild(SORT_GOODS_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          // className="button is-success is-light"
+          className={cn(
+            `button is-success
+            ${sortFild === SORT_GOODS_LENGTH || 'is-light'} `,
+          )}
+          onClick={() => setSortFild(SORT_GOODS_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          // className="button is-warning is-light"
+          className={cn(
+            `button is-warning
+            ${reversed === true || 'is-light'} `,
+          )}
+          onClick={() => setReversed(!reversed)}
+        >
+          Reverse
+        </button>
+
+        {isChoice && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => resetFild()}
```

Comment:
Let's remove unnecessary function here
```suggestion
            onClick={resetFild}
```